### PR TITLE
Fix generics schema

### PR DESCRIPTION
**Pull request by Romakita**
## Summary by CodeRabbit

* **New Features**
  * Introduced advanced support for generics in JSON schema and OpenAPI generation, including nested generics, collections, enums, and improved group handling.
  * Added new utility for merging JSON schemas, enhancing schema composition capabilities.
  * Launched a new AsyncAPI schema module with documentation and configuration files.
  * Added support for OpenAPI 3.1.0 in schema decorators and integrations.

* **Improvements**
  * Refined schema decorators and mappers for better generics, group, nullable, and reference handling.
  * Enhanced JSON schema generation and OpenAPI spec outputs with improved formatting and metadata.
  * Improved documentation formatting and clarity.
  * Simplified and standardized schema and parameter group retrieval methods.
  * Updated JSON schema internals to better encapsulate state and adopt JSON Schema draft-07 standards.
  * Enhanced schema decorator chaining and metadata application for generics and labels.
  * Streamlined middleware configuration in serverless HTTP platform integration.

* **Bug Fixes**
  * Corrected schema property handling for generics, references, required fields, and discriminator mappings.
  * Fixed issues with decorator application and property mapping in schemas.
  * Adjusted validation and error reporting paths for better accuracy.

* **Tests**
  * Expanded and updated test coverage for generics, pagination, deep object queries, discriminators, and schema merging.
  * Added comprehensive integration tests for new features and edge cases.
  * Updated test assertions to use inline snapshots for clearer spec validation.
  * Removed outdated or redundant tests related to generics and pagination.

* **Chores**
  * Cleaned up and reorganized internal code, imports, and exports for better maintainability.
  * Removed deprecated and unused code related to generics and schema utilities.
  * Simplified import paths by consolidating imports from core packages.
  * Removed redundant properties and methods in domain classes to improve encapsulation.
  * Updated package metadata and configuration files for new AsyncAPI module.

### CODE REVIEW

File: packages/specs/schema/src/components/default/genericsMapper.ts [hunk 1]
```diff
@@ -2,69 +1,97 @@
...
+      const nestedSchema = getNestedSchema(propertyKey, schema.get("items")!, options);
+
+      if (nestedSchema) {
         return {
-          type: "object"
+          type: "array",
+          items: nestedSchema as JSONSchema7
         };
       }
+    }
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/components/default/genericsMapper.ts]: in this case, check .has("items") indicate that the value is filled correctly

File: packages/specs/schema/src/components/default/requiredMapper.ts [hunk 2]
```diff
@@ -40,7 +39,7 @@ function extractRequiredProps(obj: any, schema: JsonSchema, options: JsonSchemaO
 }
 
 export function requiredMapper(obj: any, schema: JsonSchema, options: JsonSchemaOptions) {
-  if (options.groups && options.groups.includes("partial")) {
+  if (options.groups && options.groups?.includes("partial")) {
     if (obj.discriminator) {
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/components/default/requiredMapper.ts:43]: groups can have false

File: packages/specs/schema/src/decorators/common/groups.ts [hunk 3]
```diff
@@ -34,33 +31,25 @@ export function Groups<T>(groupName: string, groups: string[]): ParameterDecorat
...
+      if (groups.length == 2 && isArray(groups[1])) {
+        groupsName = groups[0];
+        groups = groups[1];
+      }
 
-        (entity as JsonParameterStore).parameter.groups = groups;
-        (entity as JsonParameterStore).parameter.groupsName = groupsName;
-        break;
+      entity.parameter.schema().groups(groups).groupsName(groupsName);
     }
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/decorators/common/groups.ts:53]: no I keep my code like that.

File: packages/specs/schema/src/domain/JsonMedia.ts [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
+      if (isArray(type)) {
+        this.schema().oneOf(type.map((type) => ({type})));
+      } else {
+        if (isClass(type)) {
+          this.schema().type("object");
+          this.schema().itemSchema(type);
+        } else {
+          this.schema().type(type);
+        }
+      }
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/domain/JsonMedia.ts:41]: this exactly what I want. itemSchema create ref to a nested schema (array, additionalProps or to a single model)

File: packages/specs/schema/src/utils/getJsonSchema.ts [hunk 5]
```diff
@@ -11,24 +11,43 @@ import {getJsonEntityStore} from "./getJsonEntityStore.js";
...
+    const entity = getJsonEntityStore(model);
+
+    let mapper = "schema";
+    if (entity instanceof JsonParameterStore) {
+      options = {
+        ...options,
+        groups: entity.schema.getGroups()
+      };
+      mapper = "item";
+    }
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/utils/getJsonSchema.ts]: No this exactly what I want!

File: packages/specs/schema/src/domain/JsonSchema.ts [hunk 6]
```diff
@@ -17,14 +33,13 @@ import {JsonEntityStore} from "./JsonEntityStore.js";
 import {JsonFormatTypes} from "./JsonFormatTypes.js";
 import {JsonLazyRef} from "./JsonLazyRef.js";
 
-export interface JsonSchemaObject extends JSONSchema6, Record<string, any> {
-  type: (any | JSONSchema6TypeName) | (any | JSONSchema6TypeName)[];
-  additionalProperties?: boolean | JSONSchema6 | any;
-  propertyNames?: boolean | JSONSchema6 | any;
-  items?: (any | JSONSchema6Definition) | (any | JSONSchema6Definition)[];
+export interface JsonSchemaObject extends Omit<JSONSchema7, "type" | "additionalProperties" | "items"> {
+  type?: Type | JSONSchema7["type"] | null | (String | null | Date | Number | Object | Boolean)[];
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/domain/JsonSchema.ts:37]: No String is intended as String class not the type

File: packages/specs/schema/test/helpers/spec-323f4cf1-e36d-4eba-bfee-8589a29a9baf.json [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  "openapi": "3.0.1",
+  "info": {
+    "title": "Title",
+    "description": "Description",
+    "termsOfService": "http://www.apache.org/",
+    "contact": {"email": "apiteam@swagger.io"},
+    "license": {"name": "Apache 2.0", "url": "http://www.apache.org/licenses/LICENSE-2.0.html"},
+    "version": "1.0.0"
+  }
+}
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/test/helpers/spec-323f4cf1-e36d-4eba-bfee-8589a29a9baf.json]: you can ignore that. it's a test

File: packages/specs/schema/src/components/default/nextMapper.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  if (schema1.canRef) {
+    const name = extraSchema.$ref.split("/").pop();
+    const titleSchema = options.components!.schemas[name];
+
+    options.components!.schemas[name] = mergeSchema(schema, titleSchema);
+
+    return extraSchema;
+  }
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/specs/schema/src/components/default/nextMapper.ts:42]: at this steps components.schemas is always defined
